### PR TITLE
Asnide 244 json parser update

### PR DESCRIPTION
**Pull request by hcorg**

### CODE REVIEW

File: src/libraries/modulemetadataparser.cpp [hunk 1]
```diff
@@ -43,35 +45,32 @@ void foreachItem(const QJsonValue &value, Func f)
...
-QString readName(const QJsonObject &object)
+QString readNotEmptyStringField(const QJsonObject &object, const QString &fieldName)
 {
-    const auto name = object["name"].toString();
-    if (name.isEmpty())
-        throw ModuleMetadataParser::Error("'name' missing or empty");
-    return name;
+    const auto value = object[fieldName].toString();
+    if (value.isEmpty())
+        throw ModuleMetadataParser::Error("'" + fieldName.toStdString() + "' midding or empty");
```
Comment (psockoN7) [inline src/libraries/modulemetadataparser.cpp]: a typo

File: src/libraries/modulemetadataparser.cpp [hunk 2]
```diff
@@ -43,35 +45,32 @@ void foreachItem(const QJsonValue &value, Func f)
...
 
-Metadata::Import readImport(const QJsonObject &object)
+QString readDescription(const QJsonObject &object)
 {
-    const auto import = Metadata::Import{object["from"].toString(),
-                                         toStringList(object["types"])};
-    if (import.from().isEmpty() || import.types().isEmpty())
-        throw ModuleMetadataParser::Error("'from' and 'types' must be present and not-empty");
-    return import;
+    return object["description"].toString();
```
Comment (psockoN7) [inline src/libraries/modulemetadataparser.cpp]: is there a reason, why readNotEmptyStringField() or readOptionalStringField() is not used?

File: src/libraries/modulemetadataparser.cpp (hunk 1 above)
Comment (hcorg) [reply src/libraries/modulemetadataparser.cpp]: fixed, thx

File: src/libraries/modulemetadataparser.cpp (hunk 2 above)
Comment (hcorg) [reply src/libraries/modulemetadataparser.cpp]: yep, it's just older code (created before those two new functions)

but true, I should refactor that

**Pull request merged**